lib/discover.js

```diff
@@ -48,7 +49,8 @@ TesselSeeker.prototype.start = function(opts) {
       .then(() => {
         debug('Connection opened:', tessel.connection.host);
         if (conn.connectionType === 'LAN' && !conn.authorized) {
-          tessel.name = conn.host.slice(0, -6);
+          // depending on the platform the returned host name could be "tessel-name.local." or "tessel-name.local"
+          tessel.name = conn.host.slice(0, conn.host.search(LOCAL_REGEX));
```

Comment:
`String#search` works not only with regexes, but strings also. You could do `.search('.local')` instead of `.search(/\.local/)`